job/service.go

```diff
@@ -303,6 +307,46 @@ func (srv *Service) Check(ctx context.Context, namespace models.NamespaceSpec, j
 	return err
 }
 
+func (srv *Service) GetJobBasicInfo(ctx context.Context, jobSpec models.JobSpec) models.JobBasicInfo {
+	var jobBasicInfo models.JobBasicInfo
+	jobBasicInfo.Spec = jobSpec
+
+	dest, err := srv.pluginService.GenerateDestination(ctx, jobSpec, jobSpec.NamespaceSpec)
+	if err != nil {
+		jobBasicInfo.Log.Write(writer.LogLevelError, fmt.Sprintf("unable to generate destination, err:%v", err))
+	}
+	if dest != nil {
+		destination := models.JobSpecTaskDestination{
+			Destination: dest.Destination,
+			Type:        dest.Type,
+		}
+		jobBasicInfo.Destination = destination.URN()
+	}
+
+	deps, err := srv.pluginService.GenerateDependencies(ctx, jobSpec, jobSpec.NamespaceSpec, false)
+	if err != nil {
+		jobBasicInfo.Log.Write(writer.LogLevelError, fmt.Sprintf("failed to generate dependencies, err:%v", err))
+	} else {
+		if deps != nil {
+			jobBasicInfo.JobSource = deps.Dependencies
+		} else {
+			jobBasicInfo.Log.Write(writer.LogLevelWarning, "no dependencies detected")
+		}
+	}
+	srv.Check(ctx, jobSpec.NamespaceSpec, []models.JobSpec{jobSpec}, &jobBasicInfo.Log)
+
+	if jobSpec.Behavior.CatchUp {
+		jobBasicInfo.Log.Write(writer.LogLevelWarning, "Catchup is enabled")
+	}
+	if dupDestJobName, err := srv.IsJobDestinationDuplicate(ctx, jobSpec); err != nil {
+		jobBasicInfo.Log.Write(writer.LogLevelError, " could not perform duplicate job destination check, err:"+err.Error())
```

Comment:
let's keep the warn/error log similar? seeing some with extra space in the front, and some using a capital letter.